### PR TITLE
Adding SASS CSS preprocessor

### PR DESCRIPTION
**Pull request by lfarran**
Adding SASS CSS preprocessor via gulp-sass and gulp-autoprefixer
Adding SASS linter with gulp-scss-lint
Updated README.md for required scss_lint Gem dependency for
gulp-sass-lint
Switched project’s CSS to SASS files and addressed SASS linting errors
Update performed with Node v5.11.1 and npm v3.8.6

Use https://github.com/toddanglin/nativescript-dev-sass for the {N} app SASS. To install: cd nativescript, tns install sass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "gulp": "gulp",
     "karma": "karma",
     "karma.start": "karma start",
-    "postinstall": "typings install && gulp check.versions && npm prune && gulp webdriver && node tools/install.js",
+    "postinstall": "typings install && gulp check.versions && npm prune && gulp webdriver && cd nativescript && npm i nativescript-dev-sass --save && cd .. && node tools/install.js",
```
Comment (NathanWalker) [inline package.json]: just revert this change. no need to modify `postinstall` since `nativescript-dev-sass` is a dependency in nativescript `package.json` now.

**Comment by lfarran** [comment]
@NathanWalker revert completed. Thanks.

**Comment by NathanWalker** [comment]
Excellent work @lfarran thank you!
